File: frontend/src/lib/types/routing.ts
import type { StationNode, TrackSegment } from './network';

export type FastestRouteResponse = {
	found: boolean;
	fromStation: string;
	toStation: string;
	trainType: string;
	path: StationNode[];
	segments: TrackSegment[];
	totalTravelMin: number;
	totalDistKm: number;
	algorithm: string;
	exploredNodesCount: number;
	message?: string | null;
};

File: backend/app/schemas/routing.py
from pydantic import BaseModel

from app.schemas.network import StationNode


class FastestRouteResult(BaseModel):
	found: bool
	fromStation: str
	toStation: str
	trainType: str
	path: list[StationNode]
	segmentIds: list[str]
	totalTravelMin: float
	totalDistKm: float
	algorithm: str = "A*"
	exploredNodesCount: int
	message: str | None = None
